### PR TITLE
Allow Hand Animations to be Overridden

### PR DESCRIPTION
**Pull request by amantley**
This changes the avatar-animation json so that we have override animations for the right and left hand, similar to how we have whole body override animations.

https://highfidelity.manuscript.com/f/cases/21958/Add-MyAvatar-overrideHandAnimation-API

Test Plan:
1) open interface in hmd mode.
2) run the following script http://hifi-content.s3.amazonaws.com/angus/handAnim/overrideHandAnimations.js
3) expected result: you should be able to trigger peace, okay, shaka, horns, and heart on the right and left hands.  The triggers differ in vive and oculus.  Vive controller inputs are very tricky to get to fire, but it should work with touching the thumb in the forward direction and the right direction.  The further trigger is to press the button on the side of the controller facing the user.  The script was developed in vive, but should work in oculus.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18418/HighFidelity-Beta-PR15271-02d9331.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18418/HighFidelity-Beta-Interface-PR15271-02d9331.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18418/HighFidelity-Beta-PR15271-02d9331.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18418/HighFidelity-Quest-Beta-PR15271-02d9331.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18418/HighFidelity-Beta-PR15271-02d9331.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18418/HighFidelity-Beta-Interface-PR15271-02d9331.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18429/HighFidelity-Beta-PR15271-d71c6c2.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18429/HighFidelity-Beta-Interface-PR15271-d71c6c2.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18429/HighFidelity-Beta-PR15271-d71c6c2.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18429/HighFidelity-Quest-Beta-PR15271-d71c6c2.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18429/HighFidelity-Beta-PR15271-d71c6c2.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18429/HighFidelity-Beta-Interface-PR15271-d71c6c2.exe

File: interface/src/avatar/MyAvatar.h [hunk 1]
```diff
@@ -576,6 +576,7 @@ class MyAvatar : public Avatar {
      * }, 3000);
      */
     Q_INVOKABLE void overrideAnimation(const QString& url, float fps, bool loop, float firstFrame, float lastFrame);
+    Q_INVOKABLE void overrideHandAnimation(bool isLeft, const QString& url, float fps, bool loop, float firstFrame, float lastFrame);
```
Comment (hyperlogic) [inline interface/src/avatar/MyAvatar.h]: Hey, please add jsdoc comments for this API call

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18475/HighFidelity-Beta-PR15271-55b4294.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18475/HighFidelity-Quest-Beta-PR15271-55b4294.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18475/HighFidelity-Beta-PR15271-55b4294.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18475/HighFidelity-Beta-Interface-PR15271-55b4294.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18475/HighFidelity-Beta-PR15271-55b4294.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18475/HighFidelity-Beta-Interface-PR15271-55b4294.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18746/HighFidelity-Beta-PR15271-ba9b539.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18746/HighFidelity-Beta-Interface-PR15271-ba9b539.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18746/HighFidelity-Beta-PR15271-ba9b539.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18746/HighFidelity-Quest-Beta-PR15271-ba9b539.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18746/HighFidelity-Beta-PR15271-ba9b539.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18746/HighFidelity-Beta-Interface-PR15271-ba9b539.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18760/HighFidelity-Beta-PR15271-035c8a6.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18760/HighFidelity-Beta-Interface-PR15271-035c8a6.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18760/HighFidelity-Beta-PR15271-035c8a6.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18760/HighFidelity-Quest-Beta-PR15271-035c8a6.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18760/HighFidelity-Beta-PR15271-035c8a6.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18760/HighFidelity-Beta-Interface-PR15271-035c8a6.exe

**Pull request merged**